src/common/TrackedOp.h

```diff
@@ -65,15 +71,26 @@ class OpHistory {
   std::atomic_bool shutdown{false};
   OpHistoryServiceThread opsvc;
   friend class OpHistoryServiceThread;
+  PerfCounters* logger;
 
 public:
-  OpHistory() : opsvc(this) {
+  OpHistory(CephContext *c) : cct(c), opsvc(this) {
+    PerfCountersBuilder b(cct, "osd-slow-op",
+                         l_osd_slow_op_first, l_osd_slow_op_last);
+    b.add_u64_counter(l_osd_slow_op_count, "slow_op",
+                      "Operation over ten second");
+
+    logger = b.create_perf_counters();
+    cct->get_perfcounters_collection()->add(logger);
+
     opsvc.create("OpHistorySvc");
   }
   ~OpHistory() {
     ceph_assert(arrived.empty());
     ceph_assert(duration.empty());
     ceph_assert(slow_op.empty());
+    cct->get_perfcounters_collection()->remove(logger);
+    delete logger;
```

Comment:
How about using `std::unique_ptr` for `logger`?

Reply:
okay